turbopack/crates/turbo-tasks-macros/src/value_impl_macro.rs

```diff
@@ -281,7 +300,12 @@ pub fn value_impl(args: TokenStream, input: TokenStream) -> TokenStream {
             #[doc(hidden)]
             #[allow(non_snake_case)]
             pub(crate) fn #register(value: &mut turbo_tasks::ValueType) {
-                value.register_trait(<Box<dyn #trait_path> as turbo_tasks::VcValueTrait>::get_trait_type_id());
+                let fat_pointer: *const dyn #trait_path = std::ptr::null::<#ty>() as *const dyn #trait_path;
+                let metadata = std::ptr::metadata(fat_pointer);
+                value.register_trait(
+                    <Box<dyn #trait_path> as turbo_tasks::VcValueTrait>::get_trait_type_id(),
+                    metadata
+                );
```

Comment:
Very tiny nit:

```suggestion
                let fat_pointer: *const dyn #trait_path = ::std::ptr::null::<#ty>() as *const dyn #trait_path;
                let metadata = ::std::ptr::metadata(fat_pointer);
                value.register_trait(
                    <::std::boxed::Box<dyn #trait_path> as turbo_tasks::VcValueTrait>::get_trait_type_id(),
                    metadata
                );
```

This reduces ambiguity.

Admittedly, we're not very consistent about this.

`turbo_tasks` cannot use the `::` prefix because of hacks we do in the `turbo_tasks` crate to `use crate as turbo_tasks`.